cadence/contracts/bridge/IFlowEVMNFTBridge.cdc

```diff
@@ -0,0 +1,115 @@
+import "FungibleToken"
+import "NonFungibleToken"
+
+import "EVM"
+
+import "FlowEVMBridgeConfig"
+import "FlowEVMBridgeUtils"
+import "CrossVMNFT"
+
+access(all) contract interface IFlowEVMNFTBridge {
+    
+    /*************
+        Events
+    **************/
+
+    /// Broadcasts an NFT was bridged from Cadence to EVM
+    access(all)
+    event BridgedNFTToEVM(
+        type: Type,
+        id: UInt64,
+        evmID: UInt256,
+        to: String,
+        evmContractAddress: String,
+        bridgeAddress: Address
+    )
+    /// Broadcasts an NFT was bridged from EVM to Cadence
+    access(all)
+    event BridgedNFTFromEVM(
+        type: Type,
+        id: UInt64,
+        evmID: UInt256,
+        caller: String,
+        evmContractAddress: String,
+        bridgeAddress: Address
+    )
+
+    /**************
+        Getters
+    ***************/
+
+    /// Returns the EVM address associated with the provided type
+    ///
+    access(all)
+    view fun getAssociatedEVMAddress(with type: Type): EVM.EVMAddress?
+
+    /// Returns the EVM address of the bridge coordinating COA
+    ///
+    access(all)
+    view fun getBridgeCOAEVMAddress(): EVM.EVMAddress
+
+    /********************************
+        Public Bridge Entrypoints
+    *********************************/
+
+    /// Public entrypoint to bridge NFTs from Cadence to EVM.
+    ///
+    /// @param token: The NFT to be bridged
+    /// @param to: The NFT recipient in FlowEVM
+    /// @param feeProvider: A reference to a FungibleToken Provider from which the bridging fee is withdrawn in $FLOW
+    ///
+    access(all)
+    fun bridgeNFTToEVM(
+        token: @{NonFungibleToken.NFT},
+        to: EVM.EVMAddress,
+        feeProvider: auth(FungibleToken.Withdraw) &{FungibleToken.Provider}
```

Comment:
Would it make sense to put a pre-condition in these to make sure the `Provider` has sufficient balance? And it has to be a `FlowToken` provider, right? Maybe we can have a pre-condition to validate that also?

Reply:
Good question. The difficulty is that when bridging an NFT to EVM, the NFT fee will be dynamically calculated based on the storage it consumes. Calculating that fee requires saving the NFT which is not a `view` method and so couldn't be added to a pre-condition